LedgerHQ/ledger-live
# feat(llm/customlockscreen): adapt logic to be more generic

### 📝 Description

- Adaptation of custom lock screen flow to be more generic
- Adaptation of device renaming for Europa device
- Removal of "Stax" hardcoded wording

**How to test:**
- Custom lock screen entry points (Stax non reg & Europa):
    - Post onboarding (not implemented for Europa yet)
    - My Ledger
    - deeplink "ledgerlive://custom-image" (LLM only)
    - Firmware update fetch & restoring (not testable for Europa yet due to unique firmware)
    - NFT
    - There should be no copy containing "Ledger Stax" instead of "Ledger Europa" (in the UI of the flow and in device actions, error messages etc.)
- Device renaming in My Ledger (Stax, Europa, Nano X):
  - It should work
  - It should not be accessible when an install/uninstall is ongoing
  - It should not be accessible at all for unsupported devices (S, SP)

**This PR does not include visual assets for Europa so for now the pictures displayed in the custom lock screen flow (for preview and transfer device action) will be using Stax assets and the pictures there will be slightly distorted. The result on the device however should match exactly the preview inside the cropping UI.**

### ❓ Context

- **JIRA or GitHub link**: [LIVE-11642] [DSDK-220] 

### ✅ Checklist

Pull Requests must pass the CI and be code reviewed. Set as Draft if the PR is not ready.

  - ...

---

### 🧐 Checklist for the PR Reviewers



- [ ] **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- [ ] **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- [ ] **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- [ ] **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- [ ] **Any new dependencies** have been justified and documented.
- [ ] **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)


[LIVE-11642]: https://ledgerhq.atlassian.net/browse/LIVE-11642?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DSDK-220]: https://ledgerhq.atlassian.net/browse/DSDK-220
